Sources/muterCore/TestReporting/Logger.swift

```diff
@@ -0,0 +1,147 @@
+import Progress
+import Foundation
+
+final class Logger {
```

Comment:
I see that much of this code was just moved from somewhere else, so feel free to ignore or defer some of my aesthetic suggestions in the interest of landing the PR if you want